src/pydap/model.py

```diff
@@ -287,18 +375,15 @@ class StructureType(DapType, Mapping):
     """A dict-like object holding other variables."""
 
     def __init__(self, name, attributes=None, **kwargs):
-        DapType.__init__(self, name, attributes, **kwargs)
+        super(StructureType, self).__init__(name, attributes, **kwargs)
 
-        # emulate a simple ordered dict
-        self._keys = []
-        self._dict = {}
+        # allow some keys to be hidden:
+        self._visible_keys = []
```

Comment:
I think this issue might have to be revisited. It is definitely not really clean. While it is cleaner now, I have the feeling that it might now belong in the ``StructureType`` class but in the ``SequenceType``. I want to be careful with this because it might come back haunt us if we want to implemented ``ConstraintExpressions`` (like the geographical constraints you inquired about) on ``StructureTypes``. At the moment, I would err on the safe side and leave it there but I want to have another look at the code, just to make sure.